### DISCUSSION TITLE
Help install SSL in my RPC

### DISCUSSION THREAD

**Discussion by Ranofty** [0xPolygon/polygon-edge#820] [Q&A]
### Your environment

* OS and version: AWS ubuntu 20.04
* version of the Polygon Edge: 0.6.0
* branch that causes this issue

Hello please i need help on how to install ssl to my json rpc any guide can help

**Comment by avenbreaks** [comment]
you can try change ip4 to dns/ and create elastic IP

**Reply by Ranofty** [reply]
> you can try change ip4 to dns/ and create elastic IP

I did that but still nothing change is still not secured do I need to run it with Nginx or any plugging?

**Reply by avenbreaks** [reply]
you can't combine nginx with libp2p, check my post 

https://github.com/0xPolygon/polygon-edge/issues/804#issuecomment-1281125440

**Reply by Ranofty** [reply]
> you can't combine nginx with libp2p, check my post
> 
> [#804 (comment)](https://github.com/0xPolygon/polygon-edge/issues/804#issuecomment-1281125440)

Thanks for the guide but can we communicate in any other platform outside GitHub? If yes drop social media name and username please will like to ask you some few questions also

**Comment by ivanbozic21** [comment]
Hi, we will move your query to the discussion board.

**Comment by laviniat1996** [comment]
Hi @Ranofty , unfortunatley this is out of the scope for our support, as this is not really a Polygon Edge related question.

**Reply by Ranofty** [reply]
Ok what did I need to do to get SSL in my RPC please can you help me with the steps in details?
Or
Do I just need to use `—dns dns/mydomain.com` instead of `—nat Public IP` or there’s anything I need to do more please I need the steps in details

also when ever I start my `nodes` even without any of this command `pm2, nohup` or any command when I just start it with normal `sudo polygon-edge server —data-dir ./mynode —chain genesis.json —libp2p 0.0.0.0:10001 —jsonrpc 0.0.0.0:80 —grpc-address 0.0.0.0:10000 —nat 3.96.142.103 —seal` it will keep on running forever i can’t stop it, restart it or do anything anymore also it’s not coming with SSL also some times the nodes will stop producing blocks but will keep on running don’t know what to do about that? 

What can I do if it stop producing blocks? 

Am waiting for your response please 🙏 
I just stuck here can’t move further without assistance 😓🙏

**Reply by laviniat1996** [reply]
@Ranofty We do not support TLS in our app directly. If you want to implement TLS, you can look into a reverse proxy like NGINX for now. We will look into this feature in the future.
